dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskInstanceServiceImpl.java

```diff
@@ -194,6 +198,24 @@ public Map<String, Object> forceTaskSuccess(User loginUser, long projectCode, In
             return result;
         }
 
+        ProcessInstance processInstance = processService.findProcessInstanceDetailById(task.getProcessInstanceId());
+        if (processInstance != null && (processInstance.getState().typeIsFailure() || processInstance.getState().typeIsCancel())) {
+            List<TaskInstance> validTaskList = processService.findValidTaskListByProcessId(processInstance.getId());
+            List<Long> instanceTaskCodeList = validTaskList.stream().map(TaskInstance::getTaskCode).collect(Collectors.toList());
+            List<ProcessTaskRelation> taskRelations = processService.findRelationByCode(processInstance.getProcessDefinitionCode(),
+                processInstance.getProcessDefinitionVersion());
+            List<TaskDefinitionLog> taskDefinitionLogs = processService.genTaskDefineList(taskRelations);
+            List<Long> definiteTaskCodeList = taskDefinitionLogs.stream().filter(definitionLog -> definitionLog.getFlag() == Flag.YES)
+                .map(TaskDefinitionLog::getCode).collect(Collectors.toList());
+            if (CollectionUtils.equalLists(instanceTaskCodeList, definiteTaskCodeList)) {
+                List<Integer> failTaskList = validTaskList.stream().filter(instance -> instance.getState().typeIsFailure() || instance.getState().typeIsCancel())
+                    .map(TaskInstance::getId).collect(Collectors.toList());
+                if (failTaskList.size() == 1 && failTaskList.contains(taskInstanceId)) {
```

Comment:
It looks strange, why is judgment = 1?

Reply:
Because the workflow should succeed only when the last failed task is forced to succeed

Reply:
I think it's better to handle processInstance after updating taskInstance, and then we don't need to use this judgment, WDYT?

Reply:
I don't think it's necessary. The update workflow and update task are in the same transaction

Reply:
It's better to update the taskInstance and then judge if the workflowInstance is success, if the workflow success, we can update the workflow status to success. 

The advantage of creating a method to judge a workflowInstance is success can be used in other place.

Reply:
I think this code is difficult to be universal